src/cosalib/qemuvariants.py

```diff
@@ -312,18 +312,21 @@ def mutate_image(self):
         elif self.compression is not None:
             sha256 = sha256sum_file(final_img)
             size = os.stat(final_img).st_size
-            temp_path = f"{final_img}.tmp"
 
+            # the compressor embeds the input filename in its output, so
+            # strip the .gz or .zip suffix from the uncompressed file
+            uncompressed_path = os.path.splitext(final_img)[0]
+            os.rename(final_img, uncompressed_path)
```

Comment:
with the

```
log.info(f"Moving {work_img} to {final_img}")
shutil.move(work_img, final_img)             
```

from above this feels a bit like musical chairs, but I don't really see a better way.